### ISSUE TITLE
Release 4.3.0 - Release Candidate 5 - Docker Listener Module testing

### ISSUE THREAD

**Issue by noise-kngdm** [wazuh/wazuh#12548]
# Docker Listener module testing

## Modules tests information
|                                  |                                             |
|----------------------------------|---------------------------------------------|
| **Main module tests issue**      | #12529  |
| **Version**                      | 4.3.0                                       |
| **Release candidate #**          | RC5                                         |
| **Tag**                          | https://github.com/wazuh/wazuh/tree/v4.3.0-rc5 |
| **Previous modules tests issue** | #12181 |

## Test report procedure

All test results must have one the following statuses: 
|                                  |                                                        |
|-----------------|-------------------------------------------------------------------------|
| :green_circle:  | The test passed successfully. 										    |
| :yellow_circle: | The test passed but some enhancements need to be applied to the module. |
| :red_circle:    | The test failed.                                                        |
| :white_circle:  | The test does not apply to the given bucket type or service.            |

Any failing test must be properly addressed with a new issue, detailing the error and the possible cause. It must be included in the [Release 4.3.0 project](https://github.com/orgs/wazuh/projects/4/) for triage.

The data available in our Wazuh-dev environment for each of these supported services will be used. Additional logs may be created if they are necessary or there is insufficient data for testing.

### Conclusions

| Test | Status Python 3.6 | Status Python 3.9 | Issue ref. |
|---|---|---|---|
| Testing docker-listener                             |    :green_circle:    |  :green_circle:   |  -  |

## Auditors validation
The definition of done for this one is the validation of the conclusions and the test results from all auditors.

All checks from below must be accepted in order to close this issue.

- [x] @davidjiglesias
- [x] @noise-kngdm
- [x] @CarlosRS9

**Comment by noise-kngdm** [comment]
## Tests

Manual testing was performed to ensure everything works after the change. The tests were carried out using both a Wazuh manager and a Wazuh agent both using the following configuration:
```
<wodle name="docker-listener">
    <interval>30s</interval>
    <attempts>5</attempts>
    <run_on_start>yes</run_on_start>
    <disabled>no</disabled>
</wodle>
```

The [logall_json option](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall-json) must be enabled in the `ossec.conf` file before starting the tests.

### Test cases Python 3.6

#### Restart Wazuh and check ossec.log to verify the module is running


<details><summary>ossec.log output</summary>

</details>


#### Test docker commands

**Run a container with no cache available**
```
docker run ubuntu
```

<details><summary>Archives.json output</summary>

</details>

**Run the same image, this time having the image in cache**

<details><summary>Archives.json output</summary>

</details>

**Remove the image**

<details><summary>Archives.json output</summary>

</details>

**Execute an application in a container**
Run `docker run -td ubuntu bash`

<details><summary>Archives.json output</summary>

</details>

**Kill the container previously launched**
Run `docker kill $(docker ps -q)`.

<details><summary>Archives.json output</summary>

</details>


### Test cases Python 3.9

#### Restart Wazuh and check ossec.log to verify the module is running


<details><summary>ossec.log output</summary>

</details>


#### Test docker commands

**Run a container with no cache available**
```
docker run ubuntu
```

<details><summary>Archives.json output</summary>

</details>

**Run the same image, this time having the image in cache**

<details><summary>Archives.json output</summary>

</details>

**Remove the image**

<details><summary>Archives.json output</summary>

</details>

**Execute an application in a container**
Run `docker run -td ubuntu bash`

<details><summary>Archives.json output</summary>

</details>

**Kill the container previously launched**
Run `docker kill $(docker ps -q)`.

<details><summary>Archives.json output</summary>

</details>
